### PR TITLE
decouple BABE from session

### LINKED ISSUE #3727: Node Template Fails to Change Epoch
When running a node template on Substrate, it fails to produce blocks after 100 due to "Expected Epoch Change":

```
2019-09-30 19:35:42 Idle (0 peers), best: #100 (0xb075…c064), finalized #100 (0xb075…c064), ⬇ 0 ⬆ 0
2019-09-30 19:35:47 Idle (0 peers), best: #100 (0xb075…c064), finalized #100 (0xb075…c064), ⬇ 0 ⬆ 0
2019-09-30 19:35:48 Starting consensus session on top of parent 0xb075eeb491f2531a81323bc2e8b9c4aa1da8f6e1d8a66bec66bd1721bbe8c064
2019-09-30 19:35:48 Prepared block for proposing at 101 [hash: 0x5e2dae479d06712d50c84baf71895bf2558ef07a189f44f9965f6ff01b2a959d; parent_hash: 0xb075…c064; extrinsics: [0xd801…18b8]]
2019-09-30 19:35:48 Pre-sealed block for proposal at 101. Hash now 0x02018d7198c46b28b3370f07ee6fbc14dfdd57f6fe7bf21aa2827a8b5326285e, previously 0x5e2dae479d06712d50c84baf71895bf2558ef07a189f44f9965f6ff01b2a959d.
2019-09-30 19:35:48 Error with block built on 0xb075eeb491f2531a81323bc2e8b9c4aa1da8f6e1d8a66bec66bd1721bbe8c064: ClientImport("Expected epoch change to happen at 0x02018d7198c46b28b3370f07ee6fbc14dfdd57f6fe7bf21aa2827a8b5326285e, s261644158")
```

Tested on Substrate commit `e0d486d617cbe24f0c91cd6aa0dfcfb6c4e4dcc2` and `7d7e74fb77b6bee2ce9d6ebafcae09caff2d0e50`

The regular Substrate node works.

@rphmeier

### PR DESCRIPTION
**Pull request by rphmeier**
Closes #3727

### CODE REVIEW

File: node-template/runtime/src/lib.rs [hunk 1]
```diff
@@ -189,6 +189,7 @@ parameter_types! {
 impl babe::Trait for Runtime {
 	type EpochDuration = EpochDuration;
 	type ExpectedBlockTime = ExpectedBlockTime;
+	type EpochChangeTrigger = babe::SameAuthoritiesForever;
```
Comment (kianenigma) [inline node-template/runtime/src/lib.rs]: Maybe put a comment here for anyone that uses node-template to be aware and change this as soon as they want to do anything more than merely _just playing around_.

File: srml/babe/src/lib.rs [hunk 2]
```diff
@@ -336,6 +369,69 @@ impl<T: Trait> Module<T> {
...
+	/// `srml-session`.
+	pub fn enact_epoch_change(
+		authorities: Vec<(AuthorityId, BabeAuthorityWeight)>,
+		next_authorities: Vec<(AuthorityId, BabeAuthorityWeight)>,
+	) {
+		// PRECONDITION: caller has done initialization and is guaranteed
+		// by the session module to be called before this.
+		#[cfg(debug_assertions)]
+		{
+			assert!(Self::initialized().is_some())
```
Comment (kianenigma) [inline srml/babe/src/lib.rs]: is this any different than `debug_assert!`? https://doc.rust-lang.org/src/core/macros.rs.html#183

Comment (rphmeier) [reply srml/babe/src/lib.rs]: this is already-committed code, but i think it's the same

**Comment by rphmeier** [comment]
(handing over to @shawntabrizi and @HCastano to see it through to merge)

**Comment by HCastano** [comment]
I managed to sync 200+ blocks with this, so it looks good!

**Comment by shawntabrizi** [comment]
+1

**Comment by illlefr4u** [comment]
for me is working too (200+ blocks)

**Pull request merged**